staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -458,3 +464,12 @@ var defaultVersionedKubernetesFeatureGates = map[featuregate.Feature]featuregate
 // defaultKubernetesFeatureGates consists of legacy unversioned Kubernetes-specific feature keys.
 // Please do not add to this struct and use defaultVersionedKubernetesFeatureGates instead.
 var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{}
+
+// testOnlyVersionedKubernetesFeatureGates consists of features that require programmatic enablement
+// for integration testing, but have not yet graduated to alpha in a release and must not be enabled
+// by a runtime option.
+var testOnlyVersionedKubernetesFeatureGates = map[featuregate.Feature]featuregate.VersionedSpecs{
+	CBORServingAndStorage: {
+		{Version: version.MustParse("1.32"), Default: false, PreRelease: featuregate.Alpha},
+	},
+}
```

Comment:
I like the idea of leveraging feature gates. This feels more intuitive to me.

Building on this idea, what if we just put the feature gate directly into `defaultKubernetesFeatureGates` but named and documented it in a clear way, maybe `TestOnlyCBORServingAndStorage` with `PreRelease: featuregate.Deprecated` 
 with a comment explaining that this is temporary and only appropriate for use in test?

Reply:
> Building on this idea, what if we just put the feature gate directly into `defaultKubernetesFeatureGates` but named and documented it in a clear way, maybe `TestOnlyCBORServingAndStorage` with `PreRelease: featuregate.Deprecated`
> with a comment explaining that this is temporary and only appropriate for use in test?

I'm hoping to prove the storage is correct and then place this into the normal gates.  The vast majority of features don't need this level of protection.  Are you ok with the approach knowing that basically only new storage encodings have ever needed this level of protection?

Reply:
> Are you ok with the approach knowing that basically only new storage encodings have ever needed this level of protection?

Yes, once this PR shifted away from exporting previously unexported variables and switched to feature gating I think I was satisfied.  The rest is just suggestions on how to communicate that the feature gates are "extra special", which, honestly, I don't feel super strong about... so long as it's reasonably clear to users what the nature of the feature gate is, I'm satisfied.